test/binary/framed/client_test.exs

```diff
@@ -22,41 +22,41 @@ defmodule BinaryFramedClientTest do
     expected_message = "Could not decode TApplicationException, remaining was <<1, 1, 1, 1>>"
     expected_type = :protocol_error
 
-    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "bad", 941, VoidReturns.MyCallResponse)
+    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "bad", 941)
     assert %TAE{message: ^expected_message,
                                          type: ^expected_type} = ex
   end
 
   thrift_test "it should be able to deserialize a message with a bad sequence id" do
     msg = <<128, 1, 0, 2, 0, 0, 0, 7, "my_call", 0, 0, 10, 197, 0>>
 
-    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 1912, VoidReturns.MyCallResponse)
+    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 1912)
     assert %TAE{type: :bad_sequence_id} = ex
   end
 
   thrift_test "it should be able to deserialize a message with the wrong method name" do
     msg = <<128, 1, 0, 2, 0, 0, 0, 8, "bad_call", 0, 0, 10, 197, 0>>
 
-    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 2757, VoidReturns.MyCallResponse)
+    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 2757)
     assert %TAE{type: :wrong_method_name} = ex
   end
 
   thrift_test "it should be able to deserialize a message with the wrong method name and sequence id " do
     msg = <<128, 1, 0, 2, 0, 0, 0, 8, "bad_call", 0, 0, 10, 197, 0>>
 
-    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 1234, VoidReturns.MyCallResponse)
-    assert %TAE{type: :sequence_id_and_rpc_name_mismatched} = ex
+    assert {:error, {:exception, ex}} = Client.deserialize_message_reply(msg, "my_call", 1234)
+    assert %TAE{type: :bad_sequence_id} = ex
   end
 
   thrift_test "it should be able to deserialize a void message" do
     msg = <<128, 1, 0, 2, 0, 0, 0, 7, "my_call", 0, 0, 10, 197, 0>>
 
-    assert {:ok, nil} =  Client.deserialize_message_reply(msg, "my_call", 2757, VoidReturns.MyCallResponse)
+    assert {:ok,  <<0>>} =  Client.deserialize_message_reply(msg, "my_call", 2757)
```

Comment:
`deserialize_message_reply` returns the serialized response struct in `:ok` responses now. It leaves it to the next layer to call `reply_module.deserialize`. It doesn't look at the bytes to know the difference between void/non-void replies.